attest/verifier/types/src/verification.rs

```diff
@@ -11,125 +17,23 @@ use mc_crypto_keys::X25519Public;
-    }
-
-    fn merge_field<B>(
-        &mut self,
-        tag: u32,
-        wire_type: WireType,
-        buf: &mut B,
-        ctx: DecodeContext,
-    ) -> Result<(), DecodeError>
-    where
-        B: Buf,
-        Self: Sized,
-    {
-        match tag {
-            TAG_DCAP_EVIDENCE_QUOTE3 => {
-                let mut vbuf = Vec::new();
-                encoding::bytes::merge(wire_type, &mut vbuf, buf, ctx)?;
-                let quote: Option<Quote3<Vec<u8>>> =
-                    mc_util_serial::deserialize(vbuf.as_slice())
-                        .map_err(|_| DecodeError::new("Failed to deserialize quote3 from bytes"))?;
-                self.quote = quote;
-                Ok(())
-            }
-            TAG_DCAP_EVIDENCE_COLLATERAL => {
-                let mut vbuf = Vec::new();
-                encoding::bytes::merge(wire_type, &mut vbuf, buf, ctx)?;
-                let collateral: Option<Collateral> = mc_util_serial::deserialize(vbuf.as_slice())
-                    .map_err(|_| {
-                    DecodeError::new("Failed to deserialize collateral from bytes")
-                })?;
-                self.collateral = collateral;
-                Ok(())
-            }
-            TAG_DCAP_EVIDENCE_REPORT_DATA => {
-                let mut vbuf = Vec::new();
-                encoding::bytes::merge(wire_type, &mut vbuf, buf, ctx)?;
-                let report_data: Option<EnclaveReportDataContents> =
-                    mc_util_serial::deserialize(vbuf.as_slice()).map_err(|_| {
-                        DecodeError::new(
-                            "Failed to deserialize EnclaveReportDataContents from bytes",
-                        )
-                    })?;
-                self.report_data = report_data;
-                Ok(())
-            }
-            _ => encoding::skip_field(wire_type, tag, buf, ctx),
-        }
-    }
-
-    fn encoded_len(&self) -> usize {
-        let quote_bytes: Vec<u8> =
-            mc_util_serial::serialize(&self.quote).expect("Failed serializing Quote3");
-        let collateral_bytes: Vec<u8> =
-            mc_util_serial::serialize(&self.collateral).expect("Failed serializing Collateral");
-        let report_data_bytes: Vec<u8> = mc_util_serial::serialize(&self.report_data)
-            .expect("Failed serializing EnclaveReportDataContents");
-
-        encoding::bytes::encoded_len(TAG_DCAP_EVIDENCE_QUOTE3, &quote_bytes)
-            + encoding::bytes::encoded_len(TAG_DCAP_EVIDENCE_COLLATERAL, &collateral_bytes)
-            + encoding::bytes::encoded_len(TAG_DCAP_EVIDENCE_REPORT_DATA, &report_data_bytes)
-    }
-
-    fn clear(&mut self) {
-        *self = Default::default();
-    }
+    pub quote: Quote3<Vec<u8>>,
+    pub collateral: Collateral,
+    pub report_data: EnclaveReportDataContents,
 }
 
 #[derive(Clone, Oneof)]
 pub enum EvidenceKind {
     #[prost(message, tag = "4")]
-    Dcap(DcapEvidence),
+    Dcap(prost::DcapEvidence),
```

Comment:
Note that this now uses the `prost::DcapEvidence`, this will force end users to convert to the standard `DcapEvidence`, but it's about the same as them having to check all 3 options that used to be there.